Libs/Optimize/ParticleSystem/itkParticleSystem.h

```diff
@@ -361,11 +361,25 @@ class ITK_EXPORT ParticleSystem : public DataObject
   /** Flag/Unflag a domain.  Flagging a domain has different meanings according
       to the application using this particle system. */
   void FlagDomain(unsigned int i)
-  { m_DomainFlags[i] = true; }
+  {
+    // ensure large enough
+    while (i >= this->m_DomainFlags.size()) {
+      m_DomainFlags.push_back(false);
+    }
+
+    // set the flag
+    m_DomainFlags[i] = true;
+  }
   void UnflagDomain(unsigned int i)
   { m_DomainFlags[i] = false; }
   bool GetDomainFlag(unsigned int i) const
-  { return m_DomainFlags[i]; }
+  {
+    if (i >= m_DomainFlags.size()) {
+      // not set
+      return false;
+    }
+    return m_DomainFlags[i];
+  }
```

Comment:
There's a m_DomainFlags here and in Optimize.h?

Reply:
There is.  It's sort of redundant.  The parameters need cleaning up for sure.  The domain vector of ints of domain flags in Optimize.h are a list of flagged domains (e.g. 2, 8, 87, 124) Whereas the particle system m_DomainFlags[i] indexes the actual domain, e.g. m_DomainFlags[2] = true, etc.